### ISSUE TITLE
Resize of scrollable table with  columnResizeMode="fit" results in columns that are not fit.

### ISSUE THREAD

**Issue by mconner** [primefaces/primeng#11364]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Stackblitz Case (Bug Reports)**

[Test Case](https://stackblitz.com/edit/primeng-tablecolresize-demo-8muwyz?file=src/app/app.component.html)
(I forked the resize demo from the 13.2.1 documentation. )

**Current behavior**

After resizing a column, a subsequent resize of the table width (e.g. by resizing the browser or iframe as in the case of the stackblitz editor), results in columns that are not fit to the table width (The screenshot shows widening the window. A scrollbar  appears if you narrow it.)

![image](https://user-images.githubusercontent.com/540402/160925223-10820716-65b1-41df-8696-39b7ef732a67.png)

**Expected behavior**

The columns should resize proportionally, as they did before resizing.  This works correctly when the table is not scrollable.

**Minimal reproduction of the problem with instructions**

1. Resize the window.
2. Columns resize to fit the width of the window, as expected in "fit" mode.
3. Resize one column. (This adds css rules for all columns.)
4. Resize the width of the window.
5. Observe that The columns do not fit the width of the table.

I believe the issue is that when scrollable, the component is adding css rules (rather than styles) for each of the columns. For example:

```
#pr_id_1-table > .p-datatable-tbody > tr > td:nth-child(1) {
    flex: 0 0 360px !important;
}

```

While that works for the current width of the table, it does not work if the table as a whole is resized, since the columns will neither grow nor shrink.  So it makes sense in "expand" columnResizeMode, but not in "fit".  It should probably be `flex: 1 1 360px !important;` or maybe `flex: 360 360 360px !important;`  to make the columns grow or shrink, proportional to their size.

**What is the motivation / use case for changing the behavior?**

Column Resize should work more or less the same whether the table is scrollable or not.

**Please tell us about your environment:**

Windows, IntelliJ

* **Angular version:** 13.x

* **PrimeNG version:** 13.2.1

* **Browser:** [Edge, Chrome]

 
* **Language:** [TypeScript 4.5.5]

**Comment by yigitfindikli** [comment]
Seems it works with the latest version of PrimeNG.

Demo: https://stackblitz.com/edit/primeng-tablecolresize-demo-hzbozv?file=src/app/app.component.html

**Comment by mconner** [comment]
Yes, it does.  I was looking at the source code for p-table last night and I realized it had been changed, and I was using an older version.    I should have used the stackblitz template specified in the new issue text, rather than starting from your documentation stackblitz, which had not been updated.  Sorry about that.  (It _would_ make it easier to write up issues if the documentation stackblitz demos were kept in sync with the releases, but I suppose that would be a fair amount of work.)

There is still a related problem:  If you resize while in expand mode, and switch to fit, it is not adjusting to fit the space available.  But that is another issue, and I'll  write it up separately.
